src/sentry/data_export/endpoints/data_export.py

```diff
@@ -34,24 +34,37 @@ def post(self, request, organization):
         if not features.has("organizations:data-export", organization):
             return Response(status=404)
 
-        limit = request.data.get("limit")
-        serializer = DataExportQuerySerializer(
-            data=request.data, context={"organization": organization, "user": request.user}
-        )
+        # Get environment_id and limit if available
         try:
             environment_id = self._get_environment_id_from_request(request, organization.id)
         except Environment.DoesNotExist as error:
             return Response(error, status=400)
+        limit = request.data.get("limit")
 
+        # Validate the data export payload
+        serializer = DataExportQuerySerializer(data=request.data)
         if not serializer.is_valid():
             return Response(serializer.errors, status=400)
-
         data = serializer.validated_data
+
+        # Validate the project field, if provided
```

Comment:
i dk how strongly you feel about this but might be nice to add to the comment that `_get_projects_by_id` actually does the validating and will raise the `PermissionDenied` error since the name of the method doesn't necessarily make it obvious

Reply:
True, will do!